server/src/inmantals/server.py

```diff
@@ -196,22 +196,34 @@ def create_tmp_project(self) -> str:
 
         install_mode = module.InstallMode.master
 
-        v2_metadata_file: str = os.path.join(self.folder_path, module.ModuleV2.MODULE_FILE)
-        v1_metadata_file: str = os.path.join(self.folder_path, module.ModuleV1.MODULE_FILE)
+        def _get_module_name():
+            module_name: Optional[str] = None
+            libs_folder = os.path.join(inmanta_project_dir, "libs")
+            if CORE_VERSION < version.Version("5"):
+                v1_metadata_file: str = os.path.join(self.folder_path, module.Module.MODULE_FILE)
+
+                if os.path.exists(v1_metadata_file):
+                    mv1 = module.Module(project=None, path=self.folder_path)
+                    module_name = mv1.name
+                    os.symlink(self.folder_path, os.path.join(libs_folder, module_name), target_is_directory=True)
+                    self.kind = module.Module
+            else:
+                v2_metadata_file: str = os.path.join(self.folder_path, module.ModuleV2.MODULE_FILE)
+                v1_metadata_file: str = os.path.join(self.folder_path, module.ModuleV1.MODULE_FILE)
 
-        module_name: Optional[str] = None
-        libs_folder = os.path.join(inmanta_project_dir, "libs")
+                if os.path.exists(v1_metadata_file):
+                    mv1 = module.ModuleV1(project=None, path=self.folder_path)
+                    module_name = mv1.name
+                    os.symlink(self.folder_path, os.path.join(libs_folder, module_name), target_is_directory=True)
+                    self.kind = module.ModuleV1
 
-        if os.path.exists(v1_metadata_file):
-            mv1 = module.ModuleV1(project=None, path=self.folder_path)
-            module_name = mv1.name
-            os.symlink(self.folder_path, os.path.join(libs_folder, module_name), target_is_directory=True)
-            self.kind = module.ModuleV1
+                elif os.path.exists(v2_metadata_file):
+                    mv2 = module.ModuleV2(project=None, path=self.folder_path, is_editable_install=True)
+                    module_name = mv2.name
+                    self.kind = module.ModuleV2
+            return module_name
 
-        elif os.path.exists(v2_metadata_file):
-            mv2 = module.ModuleV2(project=None, path=self.folder_path, is_editable_install=True)
-            module_name = mv2.name
-            self.kind = module.ModuleV2
+        module_name = _get_module_name()
```

Comment:
The `_get_module_name` method fixes a bug when using iso4 (module.ModuleV2 doesn't exist for core versions < 5)
- it keeps the same behavior as before for core versions >= 5
- for core versions < 5 we only check if a V1 module is open